### ISSUE TITLE
Questions about the code semantic

### ISSUE THREAD

**Issue by kyoungrok0517** [stanford-futuredata/ColBERT#4]
Hi. I've read your paper and now I'm trying to understand your code. Most of the part is intuitive, thanks for the readable and runnable code :). Unfortunately I'm having trouble understanding the reason to use `labels` filled with only zeros. Please understand I'm quite new to pytorch so the question might be silly.

```python
labels = torch.zeros(args.bsize, dtype=torch.long, device=DEVICE)
...
loss = criterion(out, labels[:out.size(0)])
```

**Comment by okhat** [comment]
Hi Kyoungrok! Thanks for your question.

For optimization, we treat the task as binary classification between <query, +ve passage> and <query, -ve passage>. The positive passages are placed in position zero (and the negatives in position one), hence the labels are all zeros.

**Comment by kyoungrok0517** [comment]
Thanks for the answer. As I observe `out = (bsize, 2)` and `labels = (bsize)` with zeros. As your explanation If I do `F.cross_entropy(out, labels)` then the labels evaluated only on the first column (positive samples) so we have binary classification effect? I think I'm lack of knowledge how this `cross_entropy` function works.

**Comment by okhat** [comment]
It's just standard [cross-entropy with softmax](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html), over two classes, with the labels corresponding to the positive passages.

So, yes, this is binary classification between the positive and negative examples. Notice this is different from pointwise binary classification (namely, classifying each passage as positive or negative in isolation).
